backend/api.py

```
@@ -1336,44 +1336,45 @@ def download_molecules():
-                return jsonify({'success': False, 'error': 'count must be an integer'}), 400
-            count = min(max(count, 1), 100000)  # Clamp to reasonable max
-        
-        # Validate names
-        if names:
-            if not isinstance(names, list):
-                return jsonify({'success': False, 'error': 'names must be a list'}), 400
-            if len(names) > 200:
+        MAX_COUNT = 1000
+        try:
+            count_int = int(count) if count is not None else None
+        except Exception:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)